src/app/modules/item/components/permissions-edit-dialog/permissions-edit-dialog.component.ts

```diff
@@ -0,0 +1,146 @@
+import { Component, Inject } from '@angular/core';
+import { MatDialog, MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+
+interface Permissions {
+  can_view: 'none'|'info'|'content'|'content_with_descendants'|'solution'
+  can_enter_from: boolean,
+  can_grant_view: 'none'|'enter'|'content'|'content_with_descendants'|'solution'|'solution_with_grant',
+  can_watch: 'none'|'result'|'answer'|'answer_with_grant',
+  can_edit: 'none'|'children'|'all'|'all_with_grant',
+  can_make_session_official: boolean,
+  is_owner: boolean,
+}
+
+export interface PermissionsEditData {
+  title: string,
+  permissions: Permissions
+}
+
+@Component({
+  selector: 'alg-permissions-edit-dialog',
+  templateUrl: './permissions-edit-dialog.component.html',
+  styleUrls: [ './permissions-edit-dialog.component.scss' ]
+})
+export class PermissionsEditDialogComponent {
+
+  readonly canViewValues = [
```

Comment:
In my opinion, all these texts should be in the html, not here. Actually, the array could just be copy-pasted in the html.